scex-core/src/test/scala/com/avsystem/scex/compiler/JavaScexCompilerTest.scala

```diff
@@ -1,31 +1,95 @@
   override protected def createCompiler = new XmlFriendlyJavaScexCompiler(new ScexSettings)
 
   test("typed variables test") {
-    val acl = PredefinedAccessSpecs.basicOperations
-    val expr = "#someDouble.toDegrees"
-    val context = SimpleContext(())
-    context.setTypedVariable("someDouble", math.Pi)
+    // given a valid type variable & an expression that uses the typed variable
+    val variableName = "someDouble"
+    val expr = s"#$variableName.toDegrees"
 
-    val cexpr = compiler.buildExpression
+    // then expression is compiled without errors
+    val compiledExpr = compileExpression(expr, variableName)
+
+    // when expression is evaluated with valid value in context
+    val context = initContextWithTypedVariable(variableName, math.Pi)
+    val result = compiledExpr(context)
+
+    // then the result is correct
+    assert(math.Pi.toDegrees == result)
+  }
+
+  TypedVariables.IllegalCharsInVarName.foreach { char =>
+    test(s"typed variables fail if char [code=${char.toInt}, escaped=${escapedChar(char)}] is in var name test") {
+      // given a typed variable with illegal char in name & an expression that uses the typed variable
+      val variableName = s"2varName$char"
+      val expr = s"#`$variableName`.toDegrees"
+
+      // then compilation should fail
+      withClue(s"[code=${char.toInt}] character in variable name [$variableName] should not be allowed") {
+        assertThrows[CompilationFailedException] {
+          compileExpression(expr, variableName)
+        }
+      }
+    }
+  }
+
+  SuspiciousCharsAllowedInVariableNames.foreach { char =>
+    test(s"typed variables work if char [code=$char.toInt] is in var name test") {
+      // given a typed variable with valid name & an expression that uses the typed variable
+      val variableName = s"""2varName$char"""
+      val expr = s"#`$variableName`.toDegrees"
+
+      // then expression is compiled without errors
+      val compiledExpr = compileExpression(expr, variableName)
+
+      // when expression is evaluated with valid value in context
+      val context = initContextWithTypedVariable(variableName, math.Pi)
+      val result = compiledExpr(context)
+
+      // then the result is correct
+      assert(math.Pi.toDegrees == result)
+    }
+  }
+
+  private def initContextWithTypedVariable(
+                                            variableName: String,
+                                            value: Double
+                                          ): SimpleContext[Unit] = {
+    val ctx = SimpleContext(())
+    ctx.setTypedVariable(variableName, value)
+    ctx
+  }
+
+  private def compileExpression(
+                                 expr: String,
+                                 variableName: String,
+                                 acl: List[MemberAccessSpec] = PredefinedAccessSpecs.basicOperations
+                               ): Expression[SimpleContext[Unit], Double] =
+    compiler.buildExpression
```

Comment:
Eh, there is no proper formatter config commited, please reformat it manually to be consistent with other files (2 space indentation)

Reply:
Done